iceoryx_posh/test/moduletests/test_popo_unique_port_id.cpp

```diff
@@ -26,160 +28,155 @@ using namespace ::testing;
+    ASSERT_TRUE(detectedError.has_value());
+    ASSERT_TRUE(detectedErrorLevel.has_value());
+    EXPECT_THAT(detectedError.value(),
+                Eq(iox::Error::kPOPO__TYPED_UNIQUE_ID_ROUDI_HAS_ALREADY_DEFINED_CUSTOM_UNIQUE_ID));
+    EXPECT_THAT(detectedErrorLevel.value(), Eq(iox::ErrorLevel::SEVERE));
+
+    // reset unique RouDi ID
+    iox::popo::UniquePortId::setUniqueRouDiId(iox::roudi::DEFAULT_UNIQUE_ROUDI_ID);
 }
 
-TEST(TypedUniqueId_RouDiId, GettingTheRouDiIdWithoutSettingFails)
+TEST(UniquePortId_test, GettingTheRouDiIdWithoutSettingReturnsDefaultId)
 {
     ::testing::Test::RecordProperty("TEST_ID", "68de213f-7009-4573-8791-9f09f8ba413c");
-    bool errorHandlerCalled = false;
-    auto errorHandlerGuard = iox::ErrorHandler::setTemporaryErrorHandler(
-        [&errorHandlerCalled](const iox::Error error IOX_MAYBE_UNUSED,
-                              const std::function<void()>,
-                              const iox::ErrorLevel) { errorHandlerCalled = true; });
 
-    iox::popo::internal::getUniqueRouDiId();
-    EXPECT_TRUE(errorHandlerCalled);
+    EXPECT_THAT(iox::popo::UniquePortId::getUniqueRouDiId(), Eq(iox::roudi::DEFAULT_UNIQUE_ROUDI_ID));
 }
 
-template <typename T>
-class TypedUniqueId_test : public Test
-{
-  protected:
-    using UniqueIDType = T;
-
-    iox::cxx::GenericRAII m_uniqueRouDiId{[] { iox::popo::internal::setUniqueRouDiId(0); },
-                                          [] { iox::popo::internal::unsetUniqueRouDiId(); }};
-};
-
-using Implementations = Types<TypedUniqueId<int>, TypedUniqueId<float>>;
-
-TYPED_TEST_SUITE(TypedUniqueId_test, Implementations);
-
-
-TYPED_TEST(TypedUniqueId_test, DefaultConstructorIncrementsID)
+TEST(UniquePortId_test, DefaultConstructorIncrementsID)
 {
     ::testing::Test::RecordProperty("TEST_ID", "2912c5bb-fd6d-46b4-ab32-97cfb7018860");
-    typename TestFixture::UniqueIDType a, b;
+    UniquePortId a, b;
     EXPECT_THAT(static_cast<uint64_t>(a) + 1, Eq(static_cast<uint64_t>(b)));
 }
 
-TYPED_TEST(TypedUniqueId_test, CopyConstructorSetsSameID)
+TEST(UniquePortId_test, CopyConstructorSetsSameID)
 {
     ::testing::Test::RecordProperty("TEST_ID", "dbe1a4fd-f4fe-47e5-83ef-38a9e59afd94");
-    typename TestFixture::UniqueIDType a, b(a);
+    UniquePortId a, b(a);
     EXPECT_THAT(static_cast<uint64_t>(a), Eq(static_cast<uint64_t>(b)));
 }
 
-TYPED_TEST(TypedUniqueId_test, CopyConstructorAssignmentSetsSameID)
+TEST(UniquePortId_test, CopyConstructorAssignmentSetsSameID)
 {
     ::testing::Test::RecordProperty("TEST_ID", "31dfad29-0da0-41b8-b78f-dbeda0d7e684");
-    typename TestFixture::UniqueIDType a, b;
+    UniquePortId a, b;
     a = b;
     EXPECT_THAT(a, Eq(b));
 }
 
-TYPED_TEST(TypedUniqueId_test, MoveConstructorSetsSameID)
+TEST(UniquePortId_test, MoveConstructorSetsSameID)
 {
     ::testing::Test::RecordProperty("TEST_ID", "2abb1afa-094a-4d28-a3cc-328212ad2f7b");
-    typename TestFixture::UniqueIDType a;
+    UniquePortId a;
```

Comment:
Since we would like to not have undefined behavior and an undefined state would it make sense to test here that a moved `UniquePortId` has a defined state. This means that the uint64_t value of a moved object is INVALID_ID or something?

Reply:
I would argue that a moved `UniquePortId` does not need to be invalid, in the end it is just a `uint64_t` which is also not invalid after move. It is still in a defined state since the underlying value is not touched